01_git/rock_scissors_paper.c

```diff
@@ -0,0 +1,77 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+enum GameValue {Rock, Scissors, Paper};
+enum GameResult {Win, Draw, Lose};
+
+const char* const to_string(const enum GameValue value)
+{
+    switch(value)
+    {
+    case Rock:
+        return "Rock";
+    case Paper:
+        return "Paper";
+    case Scissors:
+        return "Scissors";
+    }
+}
+
+enum GameResult get_game_result(const enum GameValue player, const enum GameValue computer)
+{
+    if (player == computer)
+    {
+        return Draw;
+    }
+
+    if ((player == Scissors && computer == Paper) || (player == Rock && computer == Scissors) ||
+        (player == Paper && computer == Rock))
+        return Win;
+
+    return Lose;
+}
+
+enum GameValue generate_computer_move()
+{
+    return (enum GameValue) rand() % (Paper + 1);
```

Comment:
It is better to declare the enum max index explicitly in the enum, as a last value